runtime/gc_check/CheckEngine.cpp

```diff
@@ -460,6 +460,15 @@ GC_CheckEngine::checkJ9Object(J9JavaVM *javaVM, J9Object* objectPtr, J9MM_Iterat
 		}
 	}
 
+#if defined(J9VM_ARCH_X86) && defined(J9VM_ENV_DATA64)
```

Comment:
I would not prevent users to enable this check on P&Z platforms (both runtime or for core checks). You should even try and verify it does work and reports warnings at one of the 2. 

From what I can see, this check is not _on_ by default, yet, what is ok, for now.